api/build.gradle

```diff
@@ -337,219 +398,223 @@ sourceSets {
   }
 }
 
-generateApi.dependsOn validateSwagger
-ideaModule.dependsOn generateApi
-compileGeneratedJava.dependsOn generateApi
-ideaModule.dependsOn generateFireCloudClient
-compileGeneratedJava.dependsOn generateFireCloudClient
-ideaModule.dependsOn generateNotebooksClient
-compileGeneratedJava.dependsOn generateNotebooksClient
-ideaModule.dependsOn generateJiraClient
-compileGeneratedJava.dependsOn generateJiraClient
-ideaModule.dependsOn generateMandrillClient
-compileGeneratedJava.dependsOn generateMandrillClient
-ideaModule.dependsOn generateMoodleClient
-compileGeneratedJava.dependsOn generateMoodleClient
-ideaModule.dependsOn generateRdrClient
-compileGeneratedJava.dependsOn generateRdrClient
-ideaModule.dependsOn generateCaptchaClient
-compileGeneratedJava.dependsOn generateCaptchaClient
-classes.dependsOn generatedClasses
-compileJava.dependsOn compileGeneratedJava
-
-war.dependsOn compileGeneratedJava
-war.dependsOn generate_appengine_web_xml
-war.dependsOn generate_appengine_cron_yaml
+// Add dependencies to the classes builtin task from the  Java plugin.
+// "This is an aggregate task that just depends on other tasks.
+// Other plugins may attach additional compilation tasks to it."
+// https://docs.gradle.org/current/userguide/java_plugin.html
+classes.dependsOn 'generatedClasses'
+
+def swaggerCodegenTasks = ['generateWorkbenchApi', 'generateFireCloudClient', 'generateNotebooksClient',
+                           'generateJiraClient', 'generateMandrillClient', 'generateMoodleClient',
+                           'generateRdrClient', 'generateCaptchaClient']
 
+// The IDEA plugin generates files that assist the IntelliJ IDE in opening Gradle-defined projects.
+// https://docs.gradle.org/current/userguide/idea_plugin.html
+project.tasks.ideaModule.dependsOn(swaggerCodegenTasks)
+
+// Java plugin's Generated Source Set gives us the compileGeneratedJava task.
+// We need to provide it with the swagger-codegen tasks that excrete Java.
+project.tasks.compileGeneratedJava.dependsOn(swaggerCodegenTasks)
+
+project.tasks.war.dependsOn(['compileGeneratedJava', 'generate_appengine_web_xml',
+        'generate_appengine_cron_yaml'])
+
+// Clear out generated Java sorces before cleaning the project.
 clean.doFirst {
   delete("${projectDir}/$swaggerTargetFolder")
 }
 
-repositories {   // repositories for Jar's you access in your code
+repositories {
   jcenter()
 }
 
-ext {
-  mapstructVersion = '1.3.1.Final'
-}
-
+// TODO(jaycarlton): see if there's a way to subdivide this block for readability
+// To show the dependency tree, try: ./project.rb gradle dependencies --configuration compile
 dependencies {
+  annotationProcessor "org.mapstruct:mapstruct-processor:$project.ext.MAPSTRUCT_VERSION"
 
-  // To show the dependency tree, try: ./project.rb gradle dependencies --configuration compile
+  compile "ch.qos.logback:logback-classic:1.2.3"
+  // Force these dependencies to safe versions, Zendesk client depends on this.
```

Comment:
contextual placement of this comment was lost here, since it now seems to refer to all following. Delete if you don't intend to maintain a semantic grouping of dependencies

Reply:
I made three copies, but, yeah. I don't think it'll be missed.